api/v1beta1/apirule_types.go

```diff
@@ -181,3 +185,8 @@ type JwtHeader struct {
 	// +optional
 	Prefix string `json:"prefix,omitempty"`
 }
+
+// Timeout for HTTP requests in seconds. The timeout can be configured up to 3900 seconds (65 minutes).
+// +kubebuilder:validation:Minimum=1
+// +kubebuilder:validation:Maximum=3900
+type Timeout uint16 // We use unit16 instead of a time.Duration because there is a bug with duration that requires additional validation of the format. Issue: checking https://github.com/kubernetes/apiextensions-apiserver/issues/56
```

Comment:
We could probably configure it as it was done [here](https://github.com/openshift/hive/pull/1690/files)
Do you think that this is not the right approach?

I guess limiting the maximum value would be harder probably

Reply:
I actually had this as a first idea and implementation. But the max value would need an implementation in the validation. Also having the pattern validation is not so nice, since we'd only allow s, m and h. When the user adds a configuration that will not match the pattern, a regex is returned and this is not a good UX imho.

That's why considering all of this I decided to go for the simple and straight forward solution.